src/smc-webapp/project_new.cjsx

```diff
@@ -272,6 +273,33 @@ ProjectNew = (name) -> rclass
             on_error     : on_error
             switch_over  : true
 
+    keydown : (e) ->
+        if e.keyCode==13 and @refs.project_new_filename.getValue().split(".").length < 2
+            @setState(warning : true)
+            ReactDOM.findDOMNode(@refs.project_new_filename.refs.input).disabled = true
+
+    accept_file : ->
+        @create_file()
+
+    decline_file : ->
+        @setState(warning : false)
+        ReactDOM.findDOMNode(@refs.project_new_filename.refs.input).disabled = false
```

Comment:
The ReactDOM.findDOMNode(@refs.project_new_filename.refs.input)? I'm trying to get the input so I can disable the input while the warning message is open because I think it would confuse users if you can change the file name while the warning message is open. ReactDOM.findDOMNode(@refs.project_new_filename) only gets the form-group div, and not the actual input. 

There is probably another way to get the input but I'm not sure how.